### PR TITLE
Backport #8102 to fix numpy requirements

### PR DESCRIPTION
**Pull request by sklam**
as titled

### CODE REVIEW

File: buildscripts/condarecipe.local/meta.yaml [hunk 1]
```diff
@@ -42,7 +42,8 @@ requirements:
     - tbb-devel 2021.*       # [not (armv6l or armv7l or aarch64 or linux32 or ppc64le)]
   run:
     - python >=3.7
-    - numpy >=1.18,<1.23
+    # NumPy 1.22.0, 1.22.1, 1.22.2 are all broken for ufuncs, see #7756
+    - numpy >=1.18,<1.23, !=1.22.0, !=1.22.1, !=1.22.2
```
Comment (stuartarchibald) [inline buildscripts/condarecipe.local/meta.yaml:46]: The condition in `run_constrained` of ` - numpy !=1.22.0, !=1.22.1, !=1.22.2` is still present?

**Review by stuartarchibald — COMMENTED** [review]
Thanks for the patch, this looks similar to #8102 but the `run_constrained` hasn't been updated, this is noted inline.

File: buildscripts/condarecipe.local/meta.yaml (hunk 1 above)
Comment (esc) [reply buildscripts/condarecipe.local/meta.yaml:46]: I can confirm this, the entry for `run_constrained` is still present on the HEAD of #8083 cf53c138e51d56da9ecf970abe23417e8779e302

**Review by stuartarchibald — APPROVED** [review]
Thanks for the update @esc, f294ff6 with `conda render` produces:
```
  run:
    - setuptools
    - libgcc-ng >=7.5.0
    - python >=3.8,<3.9.0a0
    - numpy >=1.18,<1.23, !=1.22.0, !=1.22.1, !=1.22.2
    - llvmlite >=0.38.0rc1,<0.39
    - libstdcxx-ng >=7.5.0
  run_constrained:
    - cudatoolkit >=9.2
    - scipy >=1.0
    - tbb 2021.*
    - libopenblas !=0.3.6
    - cuda-python >=11.6
```
which looks correct.

**Pull request merged**